validate: more explicit error messages in case of indentation troubles

**Current behaviour**

The `reana-client validate` returns cryptic error message when there is a problem with `reana.yaml` validation:

```console
$ cd reana-demo-root6-roofit
$ sed -i 's,^  parameters:,   parameters:,g' reana.yaml
$ reana-client validate
==> ERROR: Something went wrong when trying to validate reana.yaml
```

This is problematic because the users would not understand what's wong.

**Expected behaviour**

Users should be given hint that there is a problem with indentation of `reana.yaml` clauses.

For example, see what `yamllint` does:

```console
$ yamllint --no-warnings reana.yaml
reana.yaml
  6:4       error    syntax error: expected <block end>, but found '<block mapping start>' (syntax)
  7:5       error    wrong indentation: expected 5 but found 4  (indentation)
  18:9      error    wrong indentation: expected 10 but found 8  (indentation)
  23:9      error    wrong indentation: expected 10 but found 8  (indentation)
```

--- a/reana_client/cli/workflow.py
+++ b/reana_client/cli/workflow.py
@@ -19,6 +19,7 @@ from jsonschema.exceptions import ValidationError
 from reana_commons.config import INTERACTIVE_SESSION_TYPES, REANA_COMPUTE_BACKENDS
 from reana_commons.errors import REANAValidationError
 from reana_commons.validation.operational_options import validate_operational_options
+import yaml
 
 from reana_client.cli.files import get_files, upload_files
 from reana_client.cli.utils import (
@@ -1004,6 +1005,15 @@ def workflow_validate(
         display_message(
             "{0} is not a valid REANA specification:\n{1}".format(
                 click.format_filename(file), e.message
+            ),
+            msg_type="error",
+        )
+        sys.exit(1)
+    except yaml.parser.ParserError as e:
+        logging.debug(traceback.format_exc())
+        display_message(
+            "{0} is not a valid YAML file:\n{1}".format(
+                click.format_filename(file, shorten=True), e
             ),
             msg_type="error",
         )

